### ISSUE TITLE
sudo/privileges elevation UX

### ISSUE THREAD

**Issue by invy** [elfmz/far2l#137]
- sudo/priveleges dialog should be either DE default (like gksu and same) or should be made using far native dialogs capabilities.
  wxDialog is the worst option, in my opinion.
- far should ask for privileges elevation before starting file editing, and not before saving.

**Comment by unxed** [comment]
about "before edit":
why? if file is accessable for read, why not to allow to start edition? maybe user whishes to save this file to another location later.

**Comment by invy** [comment]
Well, agreed.

But the information, that the file requires permissions for saving should be clearly displayed in the editor, because it looks a little bit dangerous.

If privileges elevation request would be made by far internal dialog, then maybe it's not needed then. But internal dialog must clearly state for which action permission elevation is required. If I want to save /etc/default/grub, then it must say so.

**Comment by elfmz** [comment]
1) you're free to provide any askpass application you want with SUDO_ASKPASS env variable. far2l uses own only if no default is provided.
2.1) Starting file editing may not finish with saving. 
2.2) At the save moment file permissions could be changed so sudo will not be required any or vise-verse - initially accessible path could become inaccessible in some time.
2.3) Editor has save-as possibility so you can't rely on initially selected path.
3) There're technical issues with using far internal dialog:
3.1) There'is risk of using far's functional from unexpected for initial design places
3.2) sudo requires askpass app (with appropriate output) to be used. And I can't send password to sudo via stdout - stdout is already in use by IPC. Also manual sending password to sudo instead of using askpass means sudo will be unable to re-prompt on wrong password by itself.

**Comment by invy** [comment]
1) does not work with pkexec

**Comment by elfmz** [comment]
well, didn't know about pkexec :)
But the another dialog (that is yes/no without password ) still own's - not fair to have two completely different looking dialog for approximately same thing. And asking password each time is annoying..

**Comment by unxed** [comment]
Do we need this issue still open?

**Comment by invy** [comment]
Well, nothing is decided. I would say, this issue is of course not critical, but nice to have.

**Comment by volth** [comment]
When far2l's window is showed "always on top" (many window managers have this feature, which can be enable in the menu in top-left corner of the window), sudo-dialog is shown *below* far2l's window.
And it is not visible.

**Comment by unxed** [comment]
volth: confirming this behavour

**Comment by elfmz** [comment]
fixed since backend-separation merged to master
